AdobeStockStub/Model/StubConnectionFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockStub\Model;
+
+use AdobeStock\Api\Client\AdobeStock;
+use AdobeStock\Api\Client\Http\HttpInterface;
+use Magento\AdobeStockClient\Model\ConnectionFactory;
+
+/**
+ * Used for injecting the stub HttpClient to the AdobeStock API instance.
+ */
+class StubConnectionFactory extends ConnectionFactory
```

Comment:
There is no need to extend the `ConnectionFactory`. We can simply configure the 'httpClient` constructor argument of the `ConnectionWrapper` class

Reply:
Let's actually move that constructor argument form the `ConnectionWrapper` to the original `ConnectionFactory` and configure the  original `ConnectionFactory` from the DI (will help to reduce ConnectionWrapper complexity as well as handle the `\Magento\AdobeStockClient\Model\Client\Files`)

Reply:
@sivaschenko I extended the `ConnectionFactory` because if it is rewritten thought the preference mechanism an error occurred. Because the `ConnectionWrapper` constructor waits for the `ConnectionFactory` instance and not the `StubConnectionFactory.`

I like your suggestion of moving the httpClient to the DI. Will do.

Reply:
@sivaschenko I implemented things you suggested. Current progress - file generator for the stream object. Will continue.